x/txfees/keeper/feedecorator.go

```diff
@@ -59,6 +62,43 @@ func (mfd MempoolFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate b
 		}
 	}
 
+	// Local mempool filter for improper ibc packets
+	if ctx.IsCheckTx() {
```

Comment:
Do we need to add a `&& !simulate` like the above?

Reply:
No, we want simulation to error here